src/ext/java/org/opentripplanner/ext/transmodelapi/TransmodelGraphQLPlanner.java

```diff
@@ -240,22 +240,24 @@ private RoutingRequest createRequest(DataFetchingEnvironment environment) {
         }
 
         if (hasArgument(environment, "modes")) {
-            // Map modes to comma separated list in string first to be able to reuse logic in QualifiedModeSet
-            // Remove CABLE_CAR from collection because QualifiedModeSet does not support mapping (splits on '_')
-            Set<TraverseMode> modes = new HashSet<>(environment.getArgument("modes"));
-            boolean cableCar = modes.remove(TraverseMode.CABLE_CAR);
-
-            String modesAsString = modes.isEmpty() ? "" : Joiner.on(",").join(modes);
-            if (!StringUtils.isEmpty(modesAsString)) {
-                new QualifiedModeSet(modesAsString).applyToRoutingRequest(request);
-                request.setModes(request.modes);
-            } else if (cableCar) {
-                // Clear default modes in case only cable car is selected
-                request.clearModes();
-            }
+            AtomicReference<StreetMode> accessMode = new AtomicReference<>();
+            AtomicReference<StreetMode> egressMode = new AtomicReference<>();
+            AtomicReference<StreetMode> directMode = new AtomicReference<>();
+            AtomicReference<ArrayList<TransitMode>> transitModes = new AtomicReference<>();
```

Comment:
AtomicReferences imply multithreading access, use something else to achive the callback thing.

Reply:
Changed to a simple wrapper instead, to make it clearer.